tests/upgrade/0.8.2-0.9.0/jiva/volume_test.go

```diff
@@ -72,6 +86,50 @@ var _ = Describe("[jiva] TEST VOLUME UPGRADE", func() {
 			podCount := ops.GetPodRunningCountEventually("default", upgradedLabel, replicaCount+1)
 			Expect(podCount).To(Equal(replicaCount+1), "while checking pod version")
 
+			By("verifying node stickiness after upgrade")
+			// fetching replica pods after upgrade
+			podList, err = ops.PodClient.List(metav1.ListOptions{LabelSelector: replicaLabel})
+			Expect(err).ShouldNot(HaveOccurred(), "while listing replica pod")
+			for _, pod := range podList.Items {
+				Expect(nodes).To(
+					ContainElement(pod.Spec.NodeName),
+					"while verifying node stickness of replicas",
+				)
```

Comment:
Can we make use of idiomatic maya for this?
```go
import (
  pod "github.com/openebs/maya/kubernetes/pod/v1alpha1"
)

err := pod.FromList(podList).Validate(pod.IsMatchesAnyNode(nodes))
```

I have not used the exact builder pattern but as direct methods of PodList.
The more we **innovate** on these lines, we will be able to build a better core that handles many features & future toolings.

Reply:
A very simple approach to identify the need to pushing a logic into core packages can be following:
- presence of loops in caller code
- presence of conditions in caller code